source/common/http/header_map_impl.h

```diff
@@ -207,6 +208,30 @@ class HeaderMapImpl : public virtual HeaderMap, NonCopyable {
     std::list<HeaderEntryImpl>::iterator pseudo_headers_end_;
   };
 
+  class HeaderListViewImpl : public HeaderListView {
+  public:
+    HeaderListViewImpl(const HeaderList* headers) : headers_(headers) {}
+
+    std::vector<std::reference_wrapper<const HeaderString>> keys() const {
```

Comment:
can you put these non-trivial loops in the .cc file?

Reply:
done